### PR TITLE
Makes it possible to see tiles past z transitions

### PR DESCRIPTION
**Pull request by ninjanomnom**
:cl: ninjanomnom
add: You can now see objects and turfs past the map edge borders.
/:cl:

https://puu.sh/A6EQa/057293c3f3.mp4

It's a little bit rough with the animation there since we need to snowflake the movement glide for the map edge but we can use this to allow for some interesting things. This pr is just to get the visual in.

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
holy shit the madman did it

File: code/datums/components/mirage_border.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/datum/component/mirage_border
```
Comment (ninjanomnom) [inline code/datums/components/mirage_border.dm]: This doesn't have too much reason to be a component other than organizational purposes atm but I can think of expansions on this functionality that would benefit from it being a component with access to signals.

I'm considering in the future making the component handle the border teleportation as well for example.

**Comment by arsserpentarium** [comment]
I knew, that Ninjanomnom is a wizard.Powerful one.

**Comment by fludd12** [comment]
Yesssssss
Also I may or may not have an idea for similar effects with portals

File: code/datums/components/mirage_border.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		holder.pixel_x -= world.icon_size * range
+
+/datum/component/mirage_border/Destroy()
+	QDEL_NULL(holder)
+	return ..()
+
+/datum/component/mirage_border/OnTransfer(atom/thing)
+	if(!isturf(thing))
+		stack_trace("[type] added to a [parent.type]")
+		qdel(src)
```
Comment (Cyberboss) [inline code/datums/components/mirage_border.dm]: I THINK COMPONENT_INCOMPATIBLE works here, but I'll have to double check

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Yeah change that qdel to a COMPONENT_INCOMPATIBLE return

File: code/datums/components/mirage_border.dm (hunk 2 above)
Comment (Cyberboss) [reply code/datums/components/mirage_border.dm]: also remove the stack_trace, it's logged at the /component level

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**